core/shared/platform/include/platform_api_extension.h

```diff
@@ -555,6 +622,376 @@ os_socket_set_recv_timeout(bh_socket_t socket, uint64 timeout_us);
 int
 os_socket_get_recv_timeout(bh_socket_t socket, uint64 *timeout_us);
 
+/**
+ * Enable re-use of local addresses
+ *
+ * @param socket the socket to set
+ * @param enable 1 to enable or 0 to disable
+ *
+ * @return 0 if success, -1 otherwise
+ */
+int
+os_socket_set_reuse_addr(bh_socket_t socket, bool enable);
```

Comment:
[minor] wonder I wonder if more readable would be to have `is_enabled` for both `set` and `get`?